hilti/toolchain/include/compiler/printer.h

```diff
@@ -13,26 +13,32 @@
 #include <hilti/ast/id.h>
 #include <hilti/base/util.h>
 
-namespace hilti::printer {
+namespace hilti {
 
-class Stream;
+struct Plugin;
 
-/**
- * Prints an AST as HILTI source code. This consults any installed plugin
- * `print_ast` hooks.
- */
-void print(std::ostream& out, Node* root, bool compact = false);
+namespace printer {
+
+class Stream;
 
 /**
  * Prints an AST as HILTI source code. This consults any installed plugin
  * `print_ast` hooks.
+ *
+ * @param out output stream
+ * @param root top-level node of the AST to print (which does not need to be an `ASTRoot`)
+ * @param compact if true, create a one-line representation
+ * @param user_visible if true, signal to the printer that the output is
+ * intended for user consumption, permitting it to do some visual polishing
  */
-void print(Stream& stream, Node* root); // NOLINT
+void print(std::ostream& out, Node* root, bool compact = false, bool user_visible = true);
```

Comment:
This is now pretty context-sensitive while before `compact` was more about looks. WDYT about making all these parameters required? The only call site not passing all args is `CodeGen::_getOrCreateTypeInfo` which IMO should be passing `user_visible=false` even though flipping that seems to make no difference for the test suite.